lib/util/runtime-features.js

```diff
@@ -0,0 +1,79 @@
+// @ts-check
+
+'use strict'
+
+/** @typedef {`node:${string}`} NodeModuleName */
+
+/**
+ * @param {NodeModuleName} moduleName
+ * @returns {boolean}
+ */
+function detectRuntimeFeatureByNodeModule (moduleName) {
+  try {
+    require(moduleName)
+    return true
+  } catch (err) {
+    if (err.code !== 'ERR_UNKNOWN_BUILTIN_MODULE') {
+      throw err
+    }
+    return false
+  }
+}
```

Comment:
Could it be better to return the module itself as well?

Reply:
I would actually try to avoid to have a second require cache or something like that. 

I actually think also about a check for zsdt or maybe markAsClonable. 

It could help have us to test branches which are supported by the engine.